### DISCUSSION TITLE
Go Router rebuild the whole widget

### DISCUSSION THREAD

**Discussion by saswatmoo** [csells/go_router#289] [Q&A]
Hello there, 
During development, the Go router rebuilds the whole widget tree during hot reload. Is this the default behavior or we could avoid it by doing something?

**Comment by csells** [comment]
Here's how to think about how GR works wrt HR: https://github.com/csells/go_router/discussions/201

**Reply by saswatmoo** [reply]
Hi @csells, This happened on mobile also not on the web.
